src/main/java/seedu/address/model/customer/Customer.java

```diff
@@ -85,6 +92,47 @@ public boolean isSameCustomer(Customer otherCustomer) {
             && otherCustomer.getName().equals(getName());
     }
 
+    /**
+     * Adds commission to current customer.
+     *
+     * @param commission new commission to add.
+     */
+    public void addCommission(Commission commission) {
+        requireAllNonNull(commission);
+        commissions.add(commission);
+    }
+
+    /**
+     * Returns true if a commission with the same identity as {@code commission} exists in the customer's
+     * commission list.
+     */
+    public boolean hasCommission(Commission commission) {
+        requireAllNonNull(commission);
+        return commissions.contains(commission);
+    }
+
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedCommission}.
+     * {@code target} must exist in the address book.
+     * The commission identity of {@code editedCommission} must not be the same as another existing commission in the
+     * customer's commission list.
+     */
+    public void setCommission(Commission target, Commission editedCommission) {
+        assert hasCommission(target);
+        commissions.setCommission(target, editedCommission);
+    }
+
+
+    /**
+     * Removes {@code key} from this {@code Customer}.
+     * {@code key} must exist in the customer's commission list.
+     */
+    public void removeCommission(Commission key) {
+        requireAllNonNull(key);
+        assert hasCommission(key);
```

Comment:
Same question about using `assert` statements here, should this throw an exception instead?